src/ray/core_worker/core_worker.cc

```diff
@@ -567,7 +567,10 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
 #endif
 }
 
-CoreWorker::~CoreWorker() { RAY_LOG(INFO) << "Core worker is destructed"; }
+CoreWorker::~CoreWorker() {
+  RAY_LOG(INFO) << "Core worker is destructed";
+  reference_counter_->ClearAll();
```

Comment:
Actually is there a way to clear the flyweight pool?

Reply:
I'm afraid not.

> The central repository of shared value objects is known as the flyweight factory. This component is able to locate and return a reference to an object with a given value, or insert the value if no copy was previously stored. Boost.Flyweight controls the interaction of flyweights with their factory transparently to the programmer, so that a casual user of the library need not even be concerned about the presence of such factory.   [Doc](https://www.boost.org/doc/libs/1_45_0/libs/flyweight/doc/tutorial/index.html#guide)

Besides, I understand that clear flyweight pool is the minimal fix.
But since it's a more physical component than rc counter and supposed to be transparent to the core_worker, 
I don't think clearing the flyweight pool provides better semantics in here.